tests/installation/test_executor.py

```diff
@@ -739,6 +787,60 @@ def test_executor_should_write_pep610_url_references_for_urls(
     verify_installed_distribution(tmp_venv, package, expected_url_reference)
 
 
+@pytest.mark.parametrize("cached", [False, True])
+def test_executor_should_write_pep610_url_references_for_non_wheel_urls(
+    tmp_venv: VirtualEnv,
+    pool: RepositoryPool,
+    config: Config,
+    io: BufferedIO,
+    mock_file_downloads: None,
+    mocker: MockerFixture,
+    fixture_dir: FixtureDirGetter,
+    cached: bool,
```

Comment:
```suggestion
    cached_wheel: bool,
```

Maybe, we should name it more clear because it doesn't matter if the sdist is cached, only if a generated wheel is cached or not.

Reply:
How about `is_artifact_cached` since its not only about wheels.

Reply:
Ah, I thought caching is only turned on/off for the generated wheel and the sdist is always cached, but it is also turned on/off for the sdist. In that case `cached` would have been fine, but `is_artifact_cached` is better. 😄